server/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -589,6 +623,13 @@ public void onFailure(String source, Exception e) {
                 logger.error(() -> new ParameterizedMessage("unexpected failure during [{}]", source), e);
             }
         }
+
+        @Override
+        public void clusterStatePublished(ClusterChangedEvent clusterChangedEvent) {
+            rerouteService.reroute("reroute after starting shards", prioritySupplier.get(), ActionListener.wrap(
+                r -> logger.trace("reroute after starting shards succeeded"),
+                e -> logger.debug("reroute after starting shards failed", e)));
```

Comment:
I wonder if we should log at DEBUG level here if `e instanceof FailedToCommitClusterStateException || e instanceof NotMasterException` but otherwise log at WARN level. Failing to do a delayed reroute sounds like a bug to me (unless there's a master failover).

Reply:
We already log at `ERROR` in the `BatchedRerouteService` on a failure other than no longer being a master (or `WARN` if we failed to submit the task). I think that's sufficient here.

Reply:
ok